types/d3-collection/index.d.ts

```diff
@@ -16,106 +17,463 @@ export interface Stringifiable {
 // Objects
 // ---------------------------------------------------------------------
 
-export function keys(object: { [key: string]: any }): string[];
+/**
+ * Returns an array containing the property names of the specified object (an associative array).
+ * The order of the returned array is undefined.
+ *
+ * @param object An object.
+ */
+// export function keys(object: { [key: string]: any }): string[];
 // TODO: When upgrading definitions to use TS 2.2+, use "object" data type in next line
-export function keys(object: any): string[];
+/**
+ * Returns an array containing the property names of the specified object (an associative array).
+ * The order of the returned array is undefined.
+ *
+ * @param object An object.
+ */
+export function keys(object: object): string[];
 
+/**
+ * Returns an array containing the property values of the specified object (an associative array).
+ * The order of the returned array is undefined.
+ *
+ * @param object An object.
+ */
 export function values<T>(object: { [key: string]: T }): T[];
-// TODO: When upgrading definitions to use TS 2.2+, use "object" data type in next line
-export function values(object: any): any[];
+export function values<T = any>(object: object): T[];
```

Comment:
Would it be safer to use 2.1's `keyof` support here?

```ts
declare function values<TObject extends object = any>(obj: TObject): Array<TObject[keyof TObject]>;
```

This'll avoid doing silly things like this:

```ts
const dict = { x: 1, y: 2 };

const v = values<string>(point); // oops
```